test/hotspot/jtreg/vmTestbase/nsk/jdi/ClassPrepareEvent/referenceType/refType001.java

```diff
@@ -196,10 +201,16 @@ public void run() {
                                               }
 
                                               // Check that all expected ClassPrepareEvent are received
-                                              eventsReceived = true;
-                                              for (int i = 0; i < checkedTypes.length; i++) {
-                                                   if (checkedTypes[i][2] == "0")
-                                                       eventsReceived = false;
+                                              if (!eventsReceived) {
+                                                  eventsReceived = true;
+                                                  for (int i = 0; i < checkedTypes.length; i++) {
+                                                      if (checkedTypes[i][2] == "0") {
```

Comment:
This if-statement could use a comment to explain the logic.

Reply:
Done.